language-support/hs/bindings/test/DA/Ledger/Sandbox.hs

```diff
@@ -0,0 +1,123 @@
+-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module Sandbox ( -- Run a sandbox for testing on a dynamically selected port
+    SandboxSpec(..),
+    Sandbox(port),
+    startSandbox,
+    shutdownSandbox,
+    withSandbox
+    ) where
+
+import           Control.Concurrent.Async (race)
+import           Control.Monad            (when)
+import           Control.Exception        (Exception, bracket, evaluate, onException, throw)
+import           DA.Ledger                (Port (..), unPort)
+import           Data.List                (isInfixOf)
+import           Data.List.Split          (splitOn)
+import           GHC.IO.Handle            (Handle, hGetLine)
+import           Prelude                  hiding (log)
+import           System.IO                (hFlush, stdout)
+import           System.Process           (CreateProcess (..), ProcessHandle,
+                                           StdStream (CreatePipe), createProcess, getPid,
+                                           interruptProcessGroupOf, proc, waitForProcess)
+import           System.Time.Extra        (Seconds, sleep)
+
+data SandboxSpec = SandboxSpec {dar :: String}
+
+data Sandbox = Sandbox { port :: Port, proh :: ProcessHandle }
+
+selectedPort :: Int
+selectedPort = 0 --dynamic port selection
+
+sandboxProcess :: SandboxSpec -> CreateProcess
+sandboxProcess SandboxSpec{dar} =
+    proc binary [ dar, "--port", show selectedPort]
+    where
+        binary = "ledger/sandbox/sandbox-binary"
+
+startSandboxProcess :: SandboxSpec -> IO (ProcessHandle,Maybe Handle)
```

Comment:
It is probably a good idea to use `withCreateProcess` here.

Reply:
Having separate `startSandbox` and `shutdownSandbox`, we can't use `withCreateProcess`.
But I probably should make use of the standard `cleanupProcess`.